overlord/managers_test.go

```diff
@@ -5849,3 +5862,166 @@ type: kernel`
 		"snap_try_kernel": "",
 	})
 }
+
+type gadgetUpdatesSuite struct {
+	baseMgrsSuite
+}
+
+var _ = Suite(&gadgetUpdatesSuite{})
+
+func (ms *gadgetUpdatesSuite) SetUpTest(c *C) {
+	ms.baseMgrsSuite.SetUpTest(c)
+
+	bloader := boottest.MockUC16Bootenv(bootloadertest.Mock("mock", c.MkDir()))
+	bootloader.Force(bloader)
+	ms.AddCleanup(func() { bootloader.Force(nil) })
+
+	restore := release.MockOnClassic(false)
+	ms.AddCleanup(restore)
+
+	mockServer := ms.mockStore(c)
+	ms.AddCleanup(mockServer.Close)
+
+	st := ms.o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	// setup model assertion
+	model := ms.brands.Model("can0nical", "my-model", modelDefaults, map[string]interface{}{
+		"gadget": "pi",
+	})
+	devicestatetest.SetDevice(st, &auth.DeviceState{
+		Brand:  "can0nical",
+		Model:  "my-model",
+		Serial: "serialserial",
+	})
+	err := assertstate.Add(st, model)
+	c.Assert(err, IsNil)
+}
+
+// makeMockDev puts a mock mount point under /run/mnt/{structureName}
```

Comment:
this confused me a bit, it sounds like it works on the real /run.

Maybe:

 makeMockDev mocks /dev/disk/by-label/{structureName} and /dev/disk/by-label/{structureName} under the test rootdir and for osutil.LoadMountInfo ...